### ISSUE TITLE
Very bad performance when using NNUE

### ISSUE THREAD

**Issue by benwh1** [official-stockfish/Stockfish#2989]
I'm getting very bad performance when using NNUE compared to the classical evaluation. A few days ago I built the latest version with NNUE and ran a quick 50 game match (10+0.1) between it and a non-NNUE version from a few weeks before. NNUE got 10 draws and 40 losses. I got the same results with the 97f742aaefcd and 9931db908a9b nets.

I just built the new latest version with the 112bb1c8cdb5 net and it looks like I'm still getting the same results. [Here](https://lichess.org/PjEtnSAA) is an example 3+2 game.

/proc/cpuinfo: https://hastebin.com/yidocudora.sql

**Comment by ddugovic** [comment]
Is this reproducible both with and without profile-guided optimization (PGO)?

**Comment by benwh1** [comment]
> Is this reproducible both with and without profile-guided optimization (PGO)?

I'll try a non-profiled build now and post the results soon.

**Comment by vondele** [comment]
also, are you building with `ARCH=x86-64-modern` ?

**Comment by benwh1** [comment]
> also, are you building with `ARCH=x86-64-modern` ?

Just x86-64, not modern. Is there a significant difference between them?

**Comment by vondele** [comment]
yes, quite a bit. Your cpu support modern. With the latest master that is (on my system) about 25% more nps for NNUE, and that's after a number of improvements (for x86-64 plain) very recently.

**Comment by vondele** [comment]
you need at least sources of Aug. 10th (https://github.com/official-stockfish/Stockfish/commit/21df37d7fd4dcc9b4a9c319382cc43685c0259c8) to have reasonable performance on x86-64

**Comment by benwh1** [comment]
> you need at least sources of Aug. 10th ([21df37d](https://github.com/official-stockfish/Stockfish/commit/21df37d7fd4dcc9b4a9c319382cc43685c0259c8)) to have reasonable performance on x86-64

Ah ok, that probably explains the problem then. The version where I got bad performance was from August 8th.

I just finished running a 30 game 10+0.1 match on the current master branch, and I got good results both with and without profiling. 13 wins, 12 draws, 5 losses on the unprofiled build, and 14 wins, 14 draws, 2 losses on the profiled build. I tried x86-64-modern with profiling, and NNUE got 17 wins, 7 draws, 6 losses.

Thanks for the help!
